src/relayclient/RelayClient.ts

```diff
@@ -260,4 +260,28 @@ export default class RelayClient {
       httpRequest
     }
   }
+
+  async resolveForwarder (gsnTransactionDetails: GsnTransactionDetails): Promise<Address> {
+    let forwarderAddress = gsnTransactionDetails.forwarder ?? this.config.forwarderAddress
+    if (forwarderAddress !== constants.ZERO_ADDRESS) {
+      const isTrusted = await this.contractInteractor.isTrustedForwarder(gsnTransactionDetails.to, forwarderAddress)
+      if (!isTrusted) {
+        throw new Error('The Forwarder address configured but is not trusted by the Recipient contract')
+      }
+    } else {
+      try {
+        if (this.config.verbose) {
+          console.log(`will attempt to get trusted forwarder from: ${gsnTransactionDetails.to}`)
+        }
+        forwarderAddress = await this.contractInteractor.getForwarder(gsnTransactionDetails.to)
+        if (this.config.verbose) {
+          console.log(`on-chain forwarder for: ${gsnTransactionDetails.to} is ${forwarderAddress}`)
+        }
+      } catch (e) {
+        throw new Error('No forwarder address configured and fetching from Recipient failed')
```

Comment:
Please update the error: `No forwarder address configured and no getTrustedForwarder in target contract`

Reply:
Done